kolibri/core/assets/src/api_resource.js

```diff
@@ -39,24 +42,87 @@ class Model {
    * returns, otherwise reject is called with the response object.
    */
   fetch(params = {}, force = false) {
-    if (!force && this.synced) {
-      return Promise.resolve(this.attributes);
-    }
-    this.synced = false;
-    return new Promise((resolve, reject) => {
-      // Do a fetch on the URL.
-      client({ path: this.url, params }).then((response) => {
-        // Set the retrieved Object onto the Model instance.
-        this.set(response.entity);
-        // Flag that the Model has been fetched.
-        this.synced = true;
-        // Resolve the promise with the attributes of the Model.
-        resolve(this.attributes);
-      }, (response) => {
-        logging.error('An error occurred', response);
-        reject(response);
+    const promise = new Promise((resolve, reject) => {
+      Promise.all(this.promises).then(() => {
+        if (!force && this.synced) {
```

Comment:
Move this inside, so that we don't resolve a cached fetch unless a previous fetch has already completed.